File: cpp_solutions/LinkedList/Remove_Duplicates_from_Sorted_List.cpp
/*
* Given a sorted linked list, delete all duplicates such that each element appear only once.
* For example,
* Given 1->1->2, return 1->2.
* Given 1->1->2->3->3, return 1->2->3.
*
**/

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */

// Notes: It's a SORTED List

class Solution {
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
    	if(!head) return NULL;

    	ListNode* ptr = head;
    	while(ptr && ptr->next){
    		if(ptr->val == ptr->next->val)
    		{
    			ptr->next = ptr->next->next;
    		}
    		else{
    			ptr = ptr->next;
    		}
    		
    	}

    	return head;
    }
};
};

File: python_solutions/LinkedList/Remove_Duplicates_from_Sorted_List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head is None:
            return head
        
        pNode = head
        while pNode and pNode.next:
            if pNode.val == pNode.next.val:
                pNode.next = pNode.next.next
            else:
                pNode = pNode.next
            
        return head